content/en/docs/languages/net/instrumentation.md

```diff
@@ -22,39 +35,219 @@ are covered here as well as the `System.Diagnostics` API.
+app, split out the dice rolling into a library file, which then will be
+imported as a dependency by the app file.
+
+Create the library file named `Dice.cs` and add the following code to it:
+
+```csharp
+/*Dice.cs*/
+
+public class Dice
+{
+    private int min;
+    private int max;
+
+    public Dice(int min, int max)
+    {
+        this.min = min;
+        this.max = max;
+    }
+
+    public List<int> rollTheDice(int rolls)
+    {
+        List<int> results = new List<int>();
+
+        for (int i = 0; i < rolls; i++)
+        {
+            results.Add(rollOnce());
+        }
+        
+        return results;
+    }
+
+    private int rollOnce()
+    {
+        return Random.Shared.Next(min, max + 1);
+    }
+}
 
-### Initializing tracing
+```
+
+Create the app file `DiceController.cs` and add the following code to it:
 
-There are two main ways to initialize [tracing](/docs/concepts/signals/traces/),
-depending on whether you're using a console app or something that's ASP.NET
-Core-based.
+```csharp
+/*DiceController.cs*/
+using Microsoft.AspNetCore.Mvc;
+using System.Net;
 
-#### Console app
 
-To start tracing in a console app, you need to create a tracer provider.
+public class DiceController : ControllerBase
+{
+    private ILogger<DiceController> logger;
+
+    public DiceController(ILogger<DiceController> logger)
+    {
+        this.logger = logger;
+    }
+
+    [HttpGet("/rolldice")]
+    public List<int> RollDice(string player, int? rolls)
+    {
+        if(!rolls.HasValue)
+        {
+            logger.LogError("Missing rolls parameter");
+            throw new HttpRequestException("Missing rolls parameter", null, HttpStatusCode.BadRequest);
+        }
+            
+        var result = new Dice(1, 6).rollTheDice(rolls.Value);
+
+        if (string.IsNullOrEmpty(player))
+        {
+            logger.LogInformation("Anonymous player is rolling the dice: {result}", result);
+        }
+        else
+        {
+            logger.LogInformation("{player} is rolling the dice: {result}", player, result);
+        }
+
+        return result;
+    }
+}
+
+```
+
+Replace the content of the Program.cs file with the following code:
+
+```csharp
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddControllers();
+
+var app = builder.Build();
+
+app.MapControllers();
+
+app.Run();
+```
+
+To ensure that it is working, run the application with the following command and
+open <http://localhost:8080/rolldice?rolls=12> in your web browser:
+
+```sh
+dotnet run
+```
+
+You should get a list of 12 numbers in your browser window, for example:
+
+```text
+[5,6,5,3,6,1,2,5,4,4,2,4]
+```
+
+## Manual instrumentation setup
+
+### Dependencies
+
+Install the following OpenTelemetry NuGet packages:
+
+[OpenTelemetry.Exporter.Console](https://www.nuget.org/packages/OpenTelemetry.Exporter.Console)
+
+[OpenTelemetry.Extensions.Hosting](https://www.nuget.org/packages/OpenTelemetry.Extensions.Hosting)
```

Comment:
@cijothomas similar to the PR on exporters, I assume also here the Extensions.Hosting is only for ASP.NET core?

Reply:
Hosting is for anything that can take BackgroundServices and Dependency Injection, so you can use that with console apps.